src/wms/src/main/java/org/geoserver/wms/capabilities/Capabilities_1_3_0_Transformer.java

```diff
@@ -682,10 +676,17 @@ private void handleLayers() {
             }
             handleRootCrsList(srs);
 
-            CloseableIterator<LayerInfo> layers;
-            layers = catalog.list(LayerInfo.class, filter);
+            // create layer groups filter
+            Filter lgFilter = Predicates.acceptAll();
+
+            // filter layer groups by namespace if needed
+            addNameSpaceFilterIfNeed(filter, "workspace.name");
```

Comment:
Thanks Andrea, indeed this was wrong (like really wrong) and I had an error in my test case that was not detecting this issue :disappointed:.